hls-test-utils/src/Test/Hls.hs

```diff
@@ -363,6 +365,78 @@ initialiseTestRecorder envVars = do
 -- ------------------------------------------------------------
 -- Run an HLS server testing a specific plugin
 -- ------------------------------------------------------------
+class TestRunner cont res where
```

Comment:
Why do we need this? I can see you're abstracting over some common patterns, but I would have thought we could do that with a wrapper function or similar rather than needing a typeclass?

Reply:
It provide us a way to expose the root of the vsf during the test.
Some test in ghcide test need that.

Reply:
It can not be don with `wrapper ` since it need to be exposed from the inside, I do find a nicer way to do it.

Reply:
Could we just expose it to all tests and mostly ignore it?

Reply:
I updated to use wrapper version. But not sure if it is better than typeclass version🤔